### PR TITLE
Api 283 post based searching

### PR DESCRIPTION
**Pull request by jhulbert-va**
[VA Jira: API-283](https://vajira.max.gov/browse/API-283)

Still finishing a few things, but this is my initial take on it. 

Remaining Questions:
* More Validation?
* ~Is this a breaking change (I don't think so, but I want to look more into it)~ <- plugin location is dealt with in the Dockerfile, its not configured outside of kong
* Better logging? <- Even after changing the url and query, kong still logs like its the POST request

### CODE REVIEW

File: kong/go-plugins/lighthouse-fhir-post-based-searching.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+  "github.com/Kong/go-pdk"
+  "regexp"
+  "strings"
+)
+
+// Based on the implementation outlined in the link below
+// http://hl7.org/fhir/R4/http.html#search
+
+type Config struct {
+  Something string
```
Comment (bschofield-va) [inline kong/go-plugins/lighthouse-fhir-post-based-searching.go:13]: ?

Comment (jhulbert-va) [reply kong/go-plugins/lighthouse-fhir-post-based-searching.go:13]: It required there be some sort of configuration, but the actual code doesnt really need it. I'll add a comment.

File: kong/go-plugins/lighthouse-fhir-post-based-searching.go [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  logError(kong, err)
+  requestBody, err := kong.Request.GetRawBody()
+  logError(kong, err)
+
+  // Rebuild the request
+  logError(kong, kong.ServiceRequest.SetMethod("GET"))
+
+  logError(kong, kong.ServiceRequest.SetHeader("Content-Type", "application/json"))
+
+  newRequestPath := "/" + fhirVersion + "/" + resourceType
```
Comment (bschofield-va) [inline kong/go-plugins/lighthouse-fhir-post-based-searching.go:64]: Just use the old path

**Comment by jhulbert-va** [comment]
Possible Blocker for use of Go plugins: https://libertyits.slack.com/archives/GBULYN18X/p1591983412460000

File: kong/go-plugins/lighthouse-fhir-post-based-searching.go [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+  requestPath, err := kong.Request.GetPath()
+  logError(kong, err)
+  matched, err := regexp.MatchString("r4/[A-Za-z]+/_search", requestPath)
+  logError(kong, err)
+  if (!matched) {
+    kong.Log.Info("Url invalid for post based searches.")
+    // Kong will log the full url here as it goes to the application.
+    return
+  }
+  splitUrl := strings.Split(requestPath, "/")
```
[the remark this answers is not in the document]
Comment (jhulbert-va) [reply kong/go-plugins/lighthouse-fhir-post-based-searching.go:45]: I'll deal with it better. That's not the path though. The spec says 
```POST  [base]/[type]/_search{?[parameters]{&_format=[mime-type]}}```
So `_search` is actually part of the path, not the queryParams.

**Comment by jhulbert-va** [comment]
Kongs go plugin support is in its earliest stages and not suitable for incorporation into `health-apis-kong` at this time.

**Pull request closed without merging**